### ISSUE TITLE
Rely: Allow combining output of multiple rely executables

### ISSUE THREAD

**Issue by kyldvs** [facebookexperimental/reason-native#53]
It would be a cool feature for Rely to be able to run many executables written with Rely.

Let's say library Foo sets up Rely and ends up with FooTests.exe, and library Bar ends up with BarTests.exe. Having to run a command like `FooTests.exe && BarTests.exe` is not ideal. I don't see if there are more failing tests in `Bar` if anything in `Foo` fails (or they have to exit with an okay status code), passing flags like `-u` gets weird, as we improve the reporting output the summaries will be wrong and printed separately.

It would be nice to be able to do a command that aggregates the output of many Rely built binaries: `rely path/to/FooTests.exe path/to/BarTests.exe ...`.

Maybe this would work by having an optional json (or some other) format as output from the test runner, which can then be aggregated.

We can also embed this ability in every Rely executable so that we do not need to ship `rely-cli` or something, it could look like: `path/to/FooTests.exe --aggregate-with path/to/BarTests.exe`

Thoughts?

**Comment by bandersongit** [comment]
I have a few thoughts on this:

I think a lot of this behavior could be handled in a fairly straightforward matter with the upcoming reporters API. Basically we could write a reporter that can allow itself to be plugged intro multiple test frameworks.

I think some of the changes I am doing to the representation of state in the test runner should also make it possible to expose an operation to "concatenate" two TestFramework modules together.

I think I lean slightly towards the second approach as it will be easier to maintain and also extend nicely to other custom reporters (e.g. any reporter automatically works in this case, whereas if we wrote a console reporter for the first proposal and someone wanted a junit xml reporter for that case we would have to handle that separately)

I think the exact syntax you suggest makes things kind of difficult, but I think we could get a lot of the way there with an executable that depends on each individual test library and can take command line args as to which one to run specifically.

**Comment by bandersongit** [comment]
I'm planning on implementing this soon. Basically I'm thinking about making the follow changes:

To Rely.rei add 
```reason
type testLibrary;
type combineResult = {
  run: (RunConfig.t) => unit,
  cli: (RunConfig.t) => unit,
  testLibrary: testLibrary
};
let combine: list(testLibrary) => combineResult;

module TestFramework = {
...
let toLibrary: unit => testLibrary;
}
```

and usage would look like

```
Rely.combine([
  TestFramework1.toLibary, 
  TestFramework2.toLibrary()
]).cli();
```

**Comment by kyldvs** [comment]
Looks great, I like the proposed API

**Comment by bandersongit** [comment]
This is done and should be included in the 2.0.0 release
